Source/WebCore/page/LocalFrameView.cpp

```diff
@@ -1943,6 +1945,48 @@ FixedContainerEdges LocalFrameView::fixedContainerEdges(BoxSideSet sides) const
         }
     };
 
+    auto isHiddenOrNearlyTransparent = [](const RenderBox& box) {
+        if (CheckedPtr layer = box.checkedLayer(); layer && layer->isVisibilityHiddenOrOpacityZero())
+            return true;
+        return box.opacity() < PageColorSampler::nearlyTransparentAlphaThreshold;
+    };
+
+    auto primaryBackgroundColorForRenderer = [&](const RenderElement& renderer) -> Color {
+        CheckedPtr box = dynamicDowncast<RenderBox>(renderer);
+        if (!box)
+            return { };
+
+        if (box->width() <= thinBorderWidth || box->height() <= thinBorderWidth)
+            return { };
+
+        if (isHiddenOrNearlyTransparent(*box))
+            return { };
+
+        auto& styleColor = renderer.style().backgroundColor();
+        if (!styleColor.isResolvedColor())
+            return { };
+
+        return styleColor.resolvedColor();
+    };
+
+    auto isContainerEdgeCandidate = [&](const RenderElement& renderer) {
```

Comment:
```suggestion
    auto isContainerEdgeCandidate = ](const RenderElement& renderer) {
```

this doesn't need to capture anything, right?

Reply:
It currently captures the `isHiddenOrNearlyTransparent` lambda, but I suppose I could move `isHiddenOrNearlyTransparent` out into a separate static helper and then change this from `[&]` to `[]`.

Thanks for the review!

Reply:
ah, my bad, didn't realize isHiddenOrNearlyTransparent was a variable and not a function, so either way is fine!